Shoko.Server/API/v1/Implementations/ShokoServiceImplementation/ShokoServiceImplementation.cs

```diff
@@ -938,84 +938,72 @@ public string EnableDisableImage(bool enabled, int imageID, int imageType)
     {
         try
         {
-            var imgType = (ImageEntityType)imageType;
-            int animeID = 0;
+            var imgType = (CL_ImageEntityType)imageType;
+            var animeIDs = new HashSet<int>();
 
             switch (imgType)
             {
-                case ImageEntityType.AniDB_Cover:
+                case CL_ImageEntityType.AniDB_Cover:
                     var anime = RepoFactory.AniDB_Anime.GetByAnimeID(imageID);
                     if (anime == null)
-                    {
                         return "Could not find anime";
-                    }
 
                     anime.ImageEnabled = enabled ? 1 : 0;
                     RepoFactory.AniDB_Anime.Save(anime);
                     break;
 
-                case ImageEntityType.TvDB_Banner:
+                case CL_ImageEntityType.TvDB_Banner:
                     var banner = RepoFactory.TvDB_ImageWideBanner.GetByID(imageID);
                     if (banner == null)
-                    {
                         return "Could not find image";
-                    }
 
                     banner.Enabled = enabled ? 1 : 0;
                     RepoFactory.TvDB_ImageWideBanner.Save(banner);
-                    animeID = RepoFactory.CrossRef_AniDB_TvDB.GetByTvDBID(banner.SeriesID).FirstOrDefault()?.AniDBID ?? 0;
+                    foreach (var xref in RepoFactory.CrossRef_AniDB_TvDB.GetByTvDBID(banner.SeriesID))
+                        animeIDs.Add(xref.AniDBID);
                     break;
 
-                case ImageEntityType.TvDB_Cover:
+                case CL_ImageEntityType.TvDB_Cover:
                     var poster = RepoFactory.TvDB_ImagePoster.GetByID(imageID);
                     if (poster == null)
-                    {
                         return "Could not find image";
-                    }
 
                     poster.Enabled = enabled ? 1 : 0;
                     RepoFactory.TvDB_ImagePoster.Save(poster);
-                    animeID = RepoFactory.CrossRef_AniDB_TvDB.GetByTvDBID(poster.SeriesID).FirstOrDefault()?.AniDBID ?? 0;
+                    foreach (var xref in RepoFactory.CrossRef_AniDB_TvDB.GetByTvDBID(poster.SeriesID))
+                        animeIDs.Add(xref.AniDBID);
                     break;
 
-                case ImageEntityType.TvDB_FanArt:
+                case CL_ImageEntityType.TvDB_FanArt:
```

Comment:
Why rename this? I'm against CL_ shit. We have a ton of clients. It should be more specific if it needs to be separate

Reply:
It was renamed since only v1 will be using it by the end. I also added a new ImageEntityType void of data sources mixed in, so we can use it with a separate data source field when needed. I _can_ rewrite the v1 layer to use the new enum in combination with a data source, but I didn't feel like touching the v1 (and v2) api layer more than I needed to.

Reply:
If it's just Desktop, then making it obey Desktop's naming scheme seems correct

Reply:
I moved the logic into the image utils helper.